include/deal.II/numerics/matrix_tools.h

```diff
@@ -232,23 +234,23 @@ namespace MatrixCreator
    *
    * See the general documentation of this namespace for more information.
    */
-  template <int dim, typename number, int spacedim>
+  template <int dim, int spacedim, typename number>
   void create_mass_matrix (const Mapping<dim, spacedim>       &mapping,
                            const DoFHandler<dim,spacedim>    &dof,
                            const Quadrature<dim>    &q,
                            SparseMatrix<number>     &matrix,
-                           const Function<spacedim> *const a = 0,
+                           const Function<spacedim,number> *const a = 0,
```

Comment:
Until this PR there was an instantiation of `create_mass_matrix` for `SparseMatrix<float>` and `Function<double>`. But this certainly was not used wthin deal.ii. Funny enogh `create_boundary_mass_matrix` was instantiated only for double-double.